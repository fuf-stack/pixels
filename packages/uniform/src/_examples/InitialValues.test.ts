import { describe } from 'vitest';

import storySnapshots from '@repo/storybook-config/story-snapshots';

import * as stories from './InitialValues.stories';

describe('Story Snapshots', () => {
  storySnapshots(stories);
});
