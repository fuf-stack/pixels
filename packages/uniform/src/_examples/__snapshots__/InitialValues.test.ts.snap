// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Story Snapshots > TwoInputsDebug 1`] = `
<div>
  <div
    class="flex w-full flex-row justify-between gap-6"
  >
    <form
      class="grow min-w-60"
      data-form-is-valid="true"
      data-testid=""
    >
      <div
        class="grid gap-4"
      >
        <div
          class="group flex flex-col data-[hidden=true]:hidden w-full relative justify-end data-[has-label=true]:mt-[calc(var(--heroui-font-size-small)_+_10px)] is-filled"
          data-filled="true"
          data-filled-within="true"
          data-has-elements="true"
          data-has-label="true"
          data-has-value="true"
          data-slot="base"
        >
          <div
            class="h-full flex flex-col"
            data-slot="main-wrapper"
          >
            <div
              class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-xs px-3 gap-3 border-medium border-default-200 data-[hover=true]:border-default-400 h-10 min-h-10 rounded-small !duration-150 transition-colors motion-reduce:transition-none bg-content1 group-data-[focus=true]:border-focus is-filled"
              data-slot="input-wrapper"
              style="cursor: text;"
            >
              <label
                class="absolute pointer-events-none origin-top-left shrink-0 rtl:origin-top-right subpixel-antialiased block text-foreground-500 will-change-auto !duration-200 !ease-out motion-reduce:transition-none transition-[transform,color,left,opacity,translate,scale] group-data-[filled-within=true]:text-foreground group-data-[filled-within=true]:pointer-events-auto pb-0 z-20 top-1/2 -translate-y-1/2 group-data-[filled-within=true]:start-0 start-3 end-auto text-small group-data-[filled-within=true]:-translate-y-[calc(100%_+_var(--heroui-font-size-small)/2_+_20px)] pe-2 max-w-full text-ellipsis overflow-hidden"
                data-slot="label"
                for="react-aria-react-useId-mock"
                id="react-aria-react-useId-mock"
              >
                Field 1
              </label>
              <div
                class="inline-flex w-full items-center h-full box-border"
                data-slot="inner-wrapper"
              >
                <input
                  aria-describedby="react-aria-react-useId-mock react-aria-react-useId-mock"
                  aria-labelledby="react-aria-react-useId-mock"
                  class="w-full font-normal bg-transparent !outline-solid placeholder:text-foreground-500 focus-visible:outline-solid outline-transparent data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 data-[type=color]:rounded-none file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text text-small is-filled"
                  data-filled="true"
                  data-filled-within="true"
                  data-slot="input"
                  data-testid="field1"
                  id="react-aria-react-useId-mock"
                  name="field1"
                  placeholder=" "
                  tabindex="0"
                  title=""
                  type="text"
                  value="initial"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="group flex flex-col data-[hidden=true]:hidden w-full relative justify-end data-[has-label=true]:mt-[calc(var(--heroui-font-size-small)_+_10px)] is-filled"
          data-filled="true"
          data-filled-within="true"
          data-has-elements="true"
          data-has-label="true"
          data-has-value="true"
          data-slot="base"
        >
          <div
            class="h-full flex flex-col"
            data-slot="main-wrapper"
          >
            <div
              class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-xs px-3 gap-3 border-medium border-default-200 data-[hover=true]:border-default-400 h-10 min-h-10 rounded-small !duration-150 transition-colors motion-reduce:transition-none bg-content1 group-data-[focus=true]:border-focus is-filled"
              data-slot="input-wrapper"
              style="cursor: text;"
            >
              <label
                class="absolute pointer-events-none origin-top-left shrink-0 rtl:origin-top-right subpixel-antialiased block text-foreground-500 will-change-auto !duration-200 !ease-out motion-reduce:transition-none transition-[transform,color,left,opacity,translate,scale] group-data-[filled-within=true]:text-foreground group-data-[filled-within=true]:pointer-events-auto pb-0 z-20 top-1/2 -translate-y-1/2 group-data-[filled-within=true]:start-0 start-3 end-auto text-small group-data-[filled-within=true]:-translate-y-[calc(100%_+_var(--heroui-font-size-small)/2_+_20px)] pe-2 max-w-full text-ellipsis overflow-hidden"
                data-slot="label"
                for="react-aria-react-useId-mock"
                id="react-aria-react-useId-mock"
              >
                Field 2
              </label>
              <div
                class="inline-flex w-full items-center h-full box-border"
                data-slot="inner-wrapper"
              >
                <input
                  aria-describedby="react-aria-react-useId-mock react-aria-react-useId-mock"
                  aria-labelledby="react-aria-react-useId-mock"
                  class="w-full font-normal bg-transparent !outline-solid placeholder:text-foreground-500 focus-visible:outline-solid outline-transparent data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 data-[type=color]:rounded-none file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text text-small is-filled"
                  data-filled="true"
                  data-filled-within="true"
                  data-slot="input"
                  data-testid="field2"
                  id="react-aria-react-useId-mock"
                  name="field2"
                  placeholder=" "
                  tabindex="0"
                  title=""
                  type="text"
                  value="initial"
                />
              </div>
            </div>
          </div>
        </div>
        <button
          class="group tap-highlight-transparent data-[hover=true]:opacity-hover z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover w-full"
          data-focus="true"
          data-hover="true"
          data-react-aria-pressable="true"
          data-testid="reset_button"
          tabindex="0"
          type="button"
        >
          Reset
        </button>
      </div>
      <div
        class="mt-4 flex justify-end"
      >
        <button
          aria-label="Submit form"
          class="group tap-highlight-transparent data-[hover=true]:opacity-hover z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-success data-[hover=true]:opacity-hover text-white"
          data-react-aria-pressable="true"
          data-testid="form_submit_button"
          tabindex="0"
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</div>
`;
